app/components/post_draft/draft_handler/index.ts

```diff
@@ -4,36 +4,22 @@
 import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
 import withObservables from '@nozbe/with-observables';
 import React from 'react';
-import {combineLatest, of as of$} from 'rxjs';
+import {of as of$} from 'rxjs';
 import {switchMap} from 'rxjs/operators';
 
 import {DEFAULT_SERVER_MAX_FILE_SIZE} from '@constants/post_draft';
-import {observeConfig, observeLicense} from '@queries/servers/system';
+import {observeCanUploadFiles, observeConfigIntValue, observeConfigValue} from '@queries/servers/system';
 import {isMinimumServerVersion} from '@utils/helpers';
 
 import DraftHandler from './draft_handler';
 
 import type {WithDatabaseArgs} from '@typings/database/database';
 
 const enhanced = withObservables([], ({database}: WithDatabaseArgs) => {
-    const config = observeConfig(database);
-
-    const license = observeLicense(database);
-
-    const canUploadFiles = combineLatest([config, license]).pipe(
-        switchMap(([c, l]) => of$(
-            c?.EnableFileAttachments === 'true' ||
-                (l?.IsLicensed !== 'true' && l?.Compliance !== 'true' && c?.EnableMobileFileUpload === 'true'),
-        ),
-        ),
-    );
-
-    const maxFileSize = config.pipe(
-        switchMap((cfg) => of$(parseInt(cfg?.MaxFileSize || '0', 10) || DEFAULT_SERVER_MAX_FILE_SIZE)),
-    );
-
-    const maxFileCount = config.pipe(
-        switchMap((cfg) => of$(isMinimumServerVersion(cfg?.Version || '', 6, 0) ? 10 : 5)),
+    const canUploadFiles = observeCanUploadFiles(database);
+    const maxFileSize = observeConfigIntValue(database, 'MaxFileSize', DEFAULT_SERVER_MAX_FILE_SIZE);
+    const maxFileCount = observeConfigValue(database, 'Version').pipe(
+        switchMap((v) => of$(isMinimumServerVersion(v || '', 6, 0) ? 10 : 5)),
```

Comment:
no need to check for server version below 7.1 as we will have v2 only compatible with 7.1+ as 7.1 is the ESR